### PR TITLE
Add --ignore-pull-failures to command pull

### PR DESCRIPTION
**Pull request by ulyssessouza**
**What I did**
Add --ignore-pull-failures to command pull

**Related issue**
Resolves https://github.com/docker/compose-cli/issues/1409

### CODE REVIEW

File: local/compose/pull.go [hunk 1]
```diff
@@ -64,73 +67,89 @@ func (s *composeService) Pull(ctx context.Context, project *types.Project) error
...
-				Text:   "Pulling",
-			})
-			ref, err := reference.ParseNormalizedNamed(service.Image)
+			err := s.pullServiceImage(ctx, service, info, configFile, w)
 			if err != nil {
-				return err
+				if !opts.IgnoreFailures {
+					return err
+				}
+				w.Event(progress.Event{
```
Comment (ndeloof) [inline local/compose/pull.go:75]: progress event should be also created when err == nil

Comment (ndeloof) [inline local/compose/pull.go:75]: event should be created before returning error, so user see it reported on progress before the command fails

**Review by ndeloof — APPROVED** [review]
LGTM with a minor UX change

File: local/compose/pull.go (hunk 1 above)
Comment (ulyssessouza) [reply local/compose/pull.go:75]: This is done inside `err := s.pullServiceImage(ctx, service, info, configFile, w)` just some lines before. It calls https://github.com/docker/compose-cli/pull/1454/files#diff-49c56ffdcd4d1b6c21e27379db32f411e24665476483f4901e99359d2eb83177R155

Comment (gtardif) [reply local/compose/pull.go:75]: So if all event display is done inside `pullServiceImage` (that makes sense), we should not have this event displayed here, just check the ignoreFailures option and return the err or nil ?

Comment (gtardif) [reply local/compose/pull.go:75]: OK, I understand now, this is a workaround to still display the error message when discarding the error with --ignore-failure. And symetrical to what we do with push

**Pull request merged**